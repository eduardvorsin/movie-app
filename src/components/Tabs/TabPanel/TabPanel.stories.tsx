import type { Meta, StoryObj } from '@storybook/react';
import TabPanel from './TabPanel';

const meta: Meta<typeof TabPanel> = {
	title: 'components/Tabs/TabPanel',
	tags: ['autodocs'],
	argTypes: {
		className: {
			description: 'Additional classes for TabPanel'
		},
		testId: {
			description: 'Id for testing the component',
		},
		children: {
			description: 'Contents of the TabPanel component',
		},
		label: {
			description: 'Text to display in TabButton',
		},
	},
	component: TabPanel,
	parameters: {
		docs: {
			description: {
				component: 'The TabPanel component is a container used in the Tabs component to display the contents of a specific tab or section on a web page. It allows the user to see and interact with the content associated with the selected tab, providing ease of navigation and accessibility of the interface.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof TabPanel>;

export const Default: Story = {
	args: {
		label: 'tab-label',
		children: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Officia quia illum eligendi adipisci magni, atque animi facere odit. Ex in illo obcaecati natus culpa consequuntur nobis tempore quidem rem eum.Dicta deserunt vero consequatur veritatis veniam tenetur quibusdam vitae aspernatur perspiciatis quas modi voluptatum voluptatem impedit optio soluta eum, explicabo sequi totam quisquam doloribus consectetur illum tempora. Reiciendis, accusamus corporis.',
	},
	parameters: {
		docs: {
			description: {
				story: 'Default state for TabButton',
			},
		},
	},
};
