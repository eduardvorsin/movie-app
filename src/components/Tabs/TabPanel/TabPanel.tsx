import { ReactNode } from 'react';

export type Props = {
	label: string,
	className?: string,
	testId?: string,
	children: ReactNode,
};

export default function TabPanel({
	label,
	className,
	testId,
	children,
}: Props) {
	return (
		<div
			className={`w-full ${className}`}
			data-label={label}
			data-testid={testId}
		>
			{children}
		</div>

	);
}