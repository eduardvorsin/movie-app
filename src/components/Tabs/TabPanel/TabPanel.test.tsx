import { render, screen } from '@testing-library/react';
import TabPanel from './TabPanel';

describe('TabPanel tests', () => {
	it('is rendered correctly', () => {
		render(
			<TabPanel
				testId='test-tab-panel'
				label='test-label'
			>
				test content
			</TabPanel>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-tab-panel')).toBeInTheDocument();
	});

	it('is a basic snapshot', () => {
		render(
			<TabPanel
				testId='test-tab-panel'
				label='test-label'
			>
				test content
			</TabPanel>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-tab-panel')).toMatchSnapshot();
	});
});

