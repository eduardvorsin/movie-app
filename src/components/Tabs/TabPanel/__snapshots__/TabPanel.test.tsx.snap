// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanel tests is a basic snapshot 1`] = `
<div
  data-label="test-label"
  data-testid="test-tab-panel"
>
  test content
</div>
`;
